node_modules/oae-principals/lib/test/util.js

```diff
@@ -1047,3 +1127,176 @@ var assertUpdateUserFails = module.exports.assertUpdateUserFails = function(rest
         return callback();
     });
 };
+
+/**
+ * Assert that an email address can be verified
+ *
+ * @param  {RestContext}    restCtx         The REST context of a user who can perform the request
+ * @param  {String}         email           The email address to verify
+ * @param  {String}         token           The token that can be used to verify the email address
+ * @param  {Function}       callback        Invoked when the verify operation succeeds
+ * @param  {Object}         callback.err    An error that occurred, if any
+ * @throws {AssertionError}                 Thrown if the operation is unsuccessful or the expected results of verifying an email don't hold true
+ */
+var assertVerifyEmailSucceeds = module.exports.assertVerifyEmailSucceeds = function(restCtx, email, token, callback) {
+    // Determine whether the user is anonymous. The verifyEmail endpoint will sign the user in if
+    // they are anonymous and will return the user profile if they're not
+    RestAPI.User.getMe(restCtx, function(err, me) {
+        assert.ok(!err);
+        var wasAnonymous = me.anon;
+
+        RestAPI.User.verifyEmail(restCtx, email, token, function(err, user, response) {
+            assert.ok(!err);
+
+            // If the user was anonymous they should've been redirected to the /me page
+            if (wasAnonymous) {
+                assert.strictEqual(response.headers.location, '/me');
```

Comment:
I suppose if we take away auto-authentication on email verification, we can ensure this just goes to `/` with perhaps the instruction to open up the authentication widget